### PR TITLE
fix: unexpected logic in password verification for accounts with saml credentials SQSERVICES-1567

### PR DESCRIPTION
**Pull request by agisilaos**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/SQSERVICES-1567" title="SQSERVICES-1567" target="_blank">BugSQSERVICES-1567</a>  [ios] unexpected logic in password verification for accounts with saml credentials
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

With this PR we fix the unexpected logic in password verification for accounts with saml credentials. From now on In order to decide whether a user has SAML credentials, we check sso_id.subject in the User schema (not just sso_id).

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Source/Model/User/ZMUser.m [hunk 1]
```diff
@@ -474,7 +474,12 @@ - (void)updateWithTransportData:(NSDictionary *)transportData authoritative:(BOO
     
     if ([transportData optionalDictionaryForKey:@"sso_id"] || authoritative) {
         NSDictionary *ssoData = [transportData optionalDictionaryForKey:@"sso_id"];
-        self.usesCompanyLogin = nil != ssoData;
+        NSString *subject = [ssoData optionalStringForKey:@"subject"];
+        if (subject != nil || [subject length] != 0) {
```
Comment (johnxnguyen) [inline Source/Model/User/ZMUser.m:478]: Should this be `subject != nil && [subject length] > 0` ?

**Review by johnxnguyen — COMMENTED** [review]
We can also add a test where there is no subject, in which case `usesCompanyLogin` should be false.

File: Source/Model/User/ZMUser.m (hunk 1 above)
Comment (agisilaos) [reply Source/Model/User/ZMUser.m:478]: From what we discussed with @tmspzz it should be an OR.

**Review by KaterinaWire — APPROVED** [review]

**Review by johnxnguyen — APPROVED** [review]

**Comment by agisilaos** [comment]
release patch

**Comment by zenkins** [comment]
Hello, @agisilaos! Merging PR for command **release patch** for PR #**1369**. Will create a **patch** release after merged.
Next version is 289.1.1.

**Pull request merged**